File: www/webserv.bla/py-cgi/printCGIvariables.py
#!/Users/maximilian/goinfre/.brew/bin/python3 python3

import os, cgitb, cgi

# cgitb.enable()
# cgi.test()

crlf = "\r\n"

print ("Content-type:text/html\r\n\r\n")

try:
    print("QUERY_STRING = " + os.environ["QUERY_STRING"])
    print(crlf)
    print("REQUEST_METHOD = " + os.environ["REQUEST_METHOD"])
    print(crlf)
    print("CONTENT_TYPE = " + os.environ["CONTENT_TYPE"])
    print("CONTENT_LENGTH = " + os.environ["CONTENT_LENGTH"])

    print("SCRIPT_FILENAME = " + os.environ["SCRIPT_FILENAME"])
    print("SCRIPT_NAME = " + os.environ["SCRIPT_NAME"])
    print("PATH_INFO = " + os.environ["PATH_INFO"])
    print("PATH_TRANSLATED = " + os.environ["PATH_TRANSLATED"])
    print("REQUEST_URI = " + os.environ["REQUEST_URI"])
    print("SERVER_PROTOCOL = " + os.environ["SERVER_PROTOCOL"])

    print("GATEWAY_INTERFACE = " + os.environ["GATEWAY_INTERFACE"])
    print("SERVER_SOFTWARE = " + os.environ["SERVER_SOFTWARE"])
    print("REMOTE_ADDR = " + os.environ["REMOTE_ADDR"])
    print("REMOTE_PORT = " + os.environ["REMOTE_PORT"])
    print("SERVER_ADDR = " + os.environ["SERVER_ADDR"])
    print("SERVER_PORT = " + os.environ["SERVER_PORT"])
    print("SERVER_NAME = " + os.environ["SERVER_NAME"])

    print("REDIRECT_STATUS = " + os.environ["REDIRECT_STATUS"])

except:
    print("Error occured")

File: www/webserv.bla/php-cgi/printCGIvariables.php
<?php
    header('content-type: text/plain');
    echo "QUERY_STRING = "         , $_SERVER["QUERY_STRING"]       , "\n";
    echo "REQUEST_METHOD = "       , $_SERVER["REQUEST_METHOD"]     , "\n";
    echo "CONTENT_TYPE = "         , $_SERVER["CONTENT_TYPE"]       , "\n";
    echo "CONTENT_LENGTH = "       , $_SERVER["CONTENT_LENGTH"]     , "\n";
    echo "SCRIPT_FILENAME = "      , $_SERVER["SCRIPT_FILENAME"]    , "\n";
    echo "SCRIPT_NAME = "          , $_SERVER["SCRIPT_NAME"]        , "\n";
    echo "PATH_INFO = "            , $_SERVER["PATH_INFO"]          , "\n";
    echo "PATH_TRANSLATED = "      , $_SERVER["PATH_TRANSLATED"]    , "\n";
    echo "REQUEST_URI = "          , $_SERVER["REQUEST_URI"]        , "\n";
    echo "SERVER_PROTOCOL = "      , $_SERVER["SERVER_PROTOCOL"]    , "\n";
    echo "GATEWAY_INTERFACE = "    , $_SERVER["GATEWAY_INTERFACE"]  , "\n";
    echo "SERVER_SOFTWARE = "      , $_SERVER["SERVER_SOFTWARE"]    , "\n";
    echo "REMOTE_ADDR = "          , $_SERVER["REMOTE_ADDR"]        , "\n";
    echo "REMOTE_PORT = "          , $_SERVER["REMOTE_PORT"]        , "\n";
    echo "SERVER_ADDR = "          , $_SERVER["SERVER_ADDR"]        , "\n";
    echo "SERVER_PORT = "          , $_SERVER["SERVER_PORT"]        , "\n";
    echo "SERVER_NAME = "          , $_SERVER["SERVER_NAME"]        , "\n";
    echo "REDIRECT_STATUS = "      , $_SERVER["REDIRECT_STATUS"]    , "\n";
?>






















